Firebase.gitignore

```diff
@@ -0,0 +1,44 @@
+# Firebase build and deployment files
+/firebase-debug.log
+/firebase-debug.*.log
+.firebaserc
+
+# Firebase Hosting
+/firebase.json
+*.cache
+hosting/.cache
+
+# Firebase Functions
+/node_modules/
+/functions/node_modules/
+/functions/.env
+/functions/package-lock.json
+
+# Firebase Emulators
+/firebase-*.zip
+/.firebase/
+/emulator-ui/
+
+# Logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+
+# OS-specific files
+.DS_Store
+Thumbs.db
+
+# IDE files
+.idea/
+.vscode/
+*.sublime*
+
+# Environment files (local configs)
+.env
+.env.*
+.env.local
+.env.development
+.env.production
+.env.test
+.env.staging
```

Comment:
These lines are not necessary given the `.env.*` regex above would actually catch them as well.
I'd also argue that these are not Firebase specific files given `.env` files can be used with JS, Ruby, Python, etc so we may want to remove them entirely as well